fix(kubernetes): let kubectl handle namespaces

sh0rez opened the pull request:
Instead of naively setting the default namespaces on objects that don't
have one, we now overlay `$KUBECONFIG` with a patch to set the default
namespace on the context, so that `kubectl` will do the job for us.

`kubectl` does this far more intelligent than we did, especially it does
not inject the namespace on objects that don't take one.

Fixes #190 

There is one `TODO` left in injecting the patch into `$KUBECONFIG`: https://github.com/grafana/tanka/blob/0257f10a151d100ee7d645106ff76a8b71a4eae7/pkg/kubernetes/client/kubectl.go#L68-L72

rfratto commented on pkg/kubernetes/client/kubectl.go:
```diff
@@ -49,12 +53,32 @@ func (k Kubectl) Info() (*Info, error) {
+		// TODO: handle empty $KUBECONFIG
+		if strings.HasPrefix(s, "KUBECONFIG=") {
+			env[i] = fmt.Sprintf("KUBECONFIG=%s:%s", k.nsPatch, strings.TrimPrefix(s, "KUBECONFIG="))
+		}
```
Can you handle the lack of a KUBECONFIG before this gets merged? I'm not sure how many people define KUBECONFIG (I don't, for example).

sh0rez replied:
Look at the PR description :D

This is a `TODO` I'm kept for tomorrow 'cause to late to do it now :D

sh0rez commented:
Fixed an issue with `native` always returning a diff for things on the `default` namespace.

Added tests:
- separating objects in missing namespace and ready during native diff (old functionality, broke here)
- injecting $KUBECONFIG

PTAL @rfratto

rfratto commented on pkg/kubernetes/client/exec.go:
```diff
@@ -0,0 +1,56 @@
+	// prepend namespace patch to $KUBECONFIG
+	env := newEnv(e)
+	if _, ok := env["KUBECONFIG"]; !ok {
+		env["KUBECONFIG"] = "~/.kube/config" // kubectl default
```
I'm skeptical about if this works, isn't `~` just a bash expansion feature? I think this might have to use `os.UserHomeDir()` and combining paths.

sh0rez replied:
You're right, I missed this :+1:
